src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -0,0 +1,155 @@
+import { Component, Input, OnInit, SimpleChanges } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { Attribute } from '../../entities/attribute.type';
+import { DownloadedModel } from '../../entities/downloaded-model.type';
+import { Embedding } from '../../entities/embedding.type';
+import { DataHandlerHelper } from '../../helper/data-handler-helper';
+import { SettingModals } from '../../helper/modal-helper';
+
+@Component({
+  selector: 'kern-embeddings',
+  templateUrl: './embeddings.component.html',
+  styleUrls: ['./embeddings.component.scss'],
+})
+export class EmbeddingsComponent implements OnInit {
+
+  @Input() project: Project;
+  @Input() attributesArrayTextUsableUploaded: Attribute[];
+  @Input() settingModals: SettingModals;
+  @Input() isManaged: boolean;
+  @Input() embeddings: Embedding[];
+  @Input() embeddingHandlesMap: { [key: string]: any };
+  @Input() dataHandlerHelper: DataHandlerHelper;
+  @Input() attributes: Attribute[];
+
+  downloadedModels: DownloadedModel[];
+  subscriptions$: Subscription[] = [];
+  somethingLoading: boolean = false;
+
+  constructor(private projectApolloService: ProjectApolloService) { }
+
+  ngOnInit(): void {
+    this.prepareDownloadedModels();
+  }
+
+  ngOnDestroy() {
+    this.subscriptions$.forEach((subscription) => subscription.unsubscribe());
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.checkStillLoading();
+    if (changes.embeddingHandlesMap) {
+      this.checkModelDownloaded();
+    }
+  }
+
+  prepareDownloadedModels() {
+    let downloadedModelsQuery$, downloadedModelsList$;
+    [downloadedModelsQuery$, downloadedModelsList$] = this.projectApolloService.getModelProviderInfo();
+    this.subscriptions$.push(
+      downloadedModelsList$.subscribe((downloadedModels) => {
+        this.downloadedModels = downloadedModels;
+        this.checkModelDownloaded();
+      }));
+  }
+
+  checkForceHiddenHandles() {
+    const granularity = this.settingModals.embedding.create.embeddingCreationFormGroup.get('granularity').value;
+    const attId = this.settingModals.embedding.create.embeddingCreationFormGroup.get('targetAttribute').value;
+    const suggestionList = this.embeddingHandlesMap[attId];
+    for (let element of suggestionList) {
+      element = { ...element };
```

Comment:
this copies the element every time the method is called. I think i remember we needed to copy responses from gql if we want to extend the elements (in this case forceHidden) but we don't need to do this every time imo. Doing this once in the subscription (e.g. with `jsoncopy` should do it. I also might be missing something and this is necessary here. just not 100%

Reply:
This was added because of some changes for the compilation time, but now I removed it and it still works as expected :D